service/history/stateBuilder.go

```diff
@@ -569,3 +579,64 @@ func (b *stateBuilderImpl) getTimerBuilder(event *shared.HistoryEvent) *timerBui
 	}
 	return newTimerBuilder(b.shard.GetConfig(), b.logger, timeSource)
 }
+
+func (b *stateBuilderImpl) filterTransferTasksForReset() ([]int64, []int64, []persistence.Task, error) {
```

Comment:
I think it is much safer to generate transferTasks from given mutable state rather than filtering them out from the list given by stateBuilder.

Reply:
OK